lib/embeddings/semantic_search.rb

```diff
@@ -43,28 +44,42 @@ def search_for_topics(query, page = 1)
           DiscourseAi::Embeddings::VectorRepresentations::Base.current_representation(strategy)
 
         digest = OpenSSL::Digest::SHA1.hexdigest(search_term)
-        hyde_key = build_hyde_key(digest, SiteSetting.ai_embeddings_semantic_search_hyde_model)
 
-        embedding_key =
-          build_embedding_key(
-            digest,
-            SiteSetting.ai_embeddings_semantic_search_hyde_model,
-            SiteSetting.ai_embeddings_model,
-          )
-
-        hypothetical_post =
-          Discourse
-            .cache
-            .fetch(hyde_key, expires_in: 1.week) { hypothetical_post_from(search_term) }
-
-        hypothetical_post_embedding =
-          Discourse
-            .cache
-            .fetch(embedding_key, expires_in: 1.week) { vector_rep.vector_from(hypothetical_post) }
+        search_embedding = nil
+
+        if hyde
+          hyde_key = build_hyde_key(digest, SiteSetting.ai_embeddings_semantic_search_hyde_model)
+
+          embedding_key =
+            build_embedding_key(
+              digest,
+              SiteSetting.ai_embeddings_semantic_search_hyde_model,
+              SiteSetting.ai_embeddings_model,
+            )
+
+          hypothetical_post =
+            Discourse
+              .cache
+              .fetch(hyde_key, expires_in: 1.week) { hypothetical_post_from(search_term) }
+
+          search_embedding =
+            Discourse
+              .cache
+              .fetch(embedding_key, expires_in: 1.week) do
+                vector_rep.vector_from(hypothetical_post)
+              end
+        else
+          embedding_key = build_embedding_key(digest, "", SiteSetting.ai_embeddings_model)
+
+          search_embedding =
+            Discourse
+              .cache
+              .fetch(embedding_key, expires_in: 1.week) { vector_rep.vector_from(search_term) }
+        end
 
         candidate_topic_ids =
           vector_rep.asymmetric_topics_similarity_search(
-            hypothetical_post_embedding,
+            search_embedding,
```

Comment:
Or even we can totally skip assignment and go with
```ruby
hyde ? search_embedding_with_hyde : search_embedding_without_hyde
```

Reply:
I will split it up a bit, no probs